packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -502,53 +502,69 @@ export class InterfaceVpcEndpoint extends VpcEndpoint implements IInterfaceVpcEn
    */
   private endpointSubnets(props: InterfaceVpcEndpointProps) {
     const lookupSupportedAzs = props.lookupSupportedAzs ?? false;
-    const subnetSelection = props.vpc.selectSubnets({ ...props.subnets, onePerAz: true });
+    const subnets = props.vpc.selectSubnets({ ...props.subnets, onePerAz: true }).subnets;
 
-    // If we don't have an account/region, we will not be able to do filtering on AZs since
-    // they will be undefined
-    const agnosticAcct = Token.isUnresolved(this.stack.account);
-    const agnosticRegion = Token.isUnresolved(this.stack.region);
+    // Sanity check the subnet count
+    if (subnets.length == 0) {
+      throw new Error('Cannot create a VPC Endpoint with no subnets');
+    }
+
+    // If we aren't going to lookup supported AZs we'll exit early, returning the subnetIds from the provided subnet selection
+    if (!lookupSupportedAzs) {
+      return subnets.map(s => s.subnetId);
```

Comment:
Don't do this, please. The object returned by `selectSubnets` has a `subnetIds` property just for this purpose.